### ISSUE TITLE
gdrive: separate cached/serialized data

### ISSUE THREAD

**Issue by isidentical** [iterative/dvc#5580]
In GDrive we have 2 types of credential files, one is a regular one that we export from GCP console (a json file) and the other one is the version that automatically gets created by pydrive if the `save_credentials` option is true. They have a different schema so that makes them incompatible. 

Regarding this issue, currently `GDRIVE_CREDENTIALS_DATA` environment variable can be both used for storing user credentials (these are always the 'serialized' ones) and also to store raw json credentials. @shcheklein suggested that we should separate the environment variables where we pass the serialized data and the raw data. This would allow people to use serialized *service* account credentials with `GDRIVE_CREDENTIALS_DATA` and use their raw *service* account credentials with another env var (something like `GDRIVE_SERVICE_DATA`). 

One thing to note in this serialization is that, whether we are going to implement the same in our fixture too (since it is independent). It seems to be that even though the underlying authentication libraries and formats are different they seem to be compatible (with the exception of not being represented as a service account in python level, but I think there is nothing that troubles that) 

tl;dr here is the 2 primary questions (I have a naive patch)
- How should we name the second environment variable that holds *raw credentials*
- Should we implement the same logic to the tests (which I guess it the primary user of this feature?)

**Comment by isidentical** [comment]
CC: @shcheklein @efiop (from the conversation on the #5554)

**Comment by shcheklein** [comment]
> Should we implement the same logic to the tests (which I guess it the primary user of this feature?)

I would add to this that CML (and similar cases when people run DVC in CI)  is the main user for this feature. So it makes it more important than our tests. Good catch on this credentials issue, @isidentical .

And our tests can definitely use the primary JSON file (it's easier for us to manage I think, no need to generate credentials locally?)

> How should we name the second environment variable that holds raw credentials

`GDRIVE_SERVICE_ACCOUNT_DATA` sounds reasonable to me
